### PR TITLE
WIP: Clean up Dockerfile

### PR DESCRIPTION
**Pull request by cgwalters**
Fix `make binaries` to also build `setup-etcd-environment`, then
use that in the `Dockerfile` to reduce duplication.

Similarly, just use globs to avoid hardcoding the architecture,
and also grab all binaries.

Add comments in various places.

### CODE REVIEW

File: Dockerfile.machine-config-operator [hunk 1]
```diff
@@ -1,20 +1,18 @@
+# This is the "all in one" image for the MCO; we previously
```
Comment (runcom) [inline Dockerfile.machine-config-operator]: I know it's a pain but `Dockerfile` must be updated as well till we remove this

**Comment by kikisdeliveryservice** [comment]
Thanks @cgwalters !! 
Can you copy these changes into `Dockerfile` since that will be the master mco dockerfile going forward?  (the others will be removed in my other pr). For now `Dockerfile` and `Dockerfile.machine-config-operator` are exact copies.

File: Dockerfile.machine-config-operator (hunk 1 above)
Comment (cgwalters) [reply Dockerfile.machine-config-operator]: Oh right I changed the wrong one, argh.

**Comment by cgwalters** [comment]
Can't we just symlink them?

**Comment by runcom** [comment]
> Can't we just symlink them?

we're gonna remove it soon enough - we didn't expect a change to it :smile:

**Comment by cgwalters** [comment]
https://github.com/openshift/machine-config-operator/pull/947

**Comment by kikisdeliveryservice** [comment]
> > Can't we just symlink them?
> 
> we're gonna remove it soon enough - we didn't expect a change to it smile

Basically this. :)

**Comment by cgwalters** [comment]
(The real reason I'm doing this is `make deploy-$x` is broken at the moment and I use it a lot)

**Comment by kikisdeliveryservice** [comment]
this pr will have to be rebased onto #947 and these changes will go into `Dockerfile`

File: Makefile [hunk 2]
```diff
@@ -105,3 +105,7 @@ images.rhel7: $(imc7)
 # This was copied from https://github.com/openshift/cluster-image-registry-operato
 test-e2e:
 	go test -timeout 75m -v$${WHAT:+ -run="$$WHAT"} ./test/e2e/
+
+Dockerfile.rhel7: Dockerfile Makefile
+	(echo 'THIS FILE IS GENERATED FROM '$<' DO NOT EDIT' && \
+	 sed -e s,org/openshift/release,org/ocp/builder, -e s,/openshift/origin-v4.0:base,/ocp/4.0:base, < $<) > $@.tmp && mv $@.tmp $@
```
Comment (cgwalters) [inline Makefile]: Stand back, I'm pulling out the `sed`!

**Comment by cgwalters** [comment]
Wait ugh...does the version of docker in 3.11 not support globs or something?

**Comment by runcom** [comment]
> Wait ugh...does the version of docker in 3.11 not support globs or something?

no version support glob afaict, in the Dockerfile you're not running a shell with expansions and whatnot

**Comment by cgwalters** [comment]
> no version support glob afaict, in the Dockerfile you're not running a shell with expansions and whatnot

See
https://docs.docker.com/engine/reference/builder/#copy

Specifically:

> Each <src> may contain wildcards and matching will be done using Go’s filepath.Match rules. For example:

And this works with `podman-1.3.1-1.git7210727.fc30.x86_64` locally.

**Comment by runcom** [comment]
> See
> https://docs.docker.com/engine/reference/builder/#copy

then I completely missed the feature and was left really behind, TIL

**Comment by cgwalters** [comment]
Obsoleted by https://github.com/openshift/machine-config-operator/pull/1210

**Pull request closed without merging**